src/sdk/namespace_cache.js

```diff
@@ -243,6 +243,23 @@ class NamespaceCache {
 
                 return object_info_cache;
 
+            } else if (object_info_hub.first_range_data && object_info_hub.size <= object_info_hub.first_range_data.length) {
+                // If the inline read covers entire object, we will submit it to cache.
+                _global_cache_uploader.submit_background(
+                    object_info_hub.size,
+                    async () => {
+                        const upload_params = {
+                            source_stream: buffer_utils.buffer_to_read_stream(object_info_hub.first_range_data),
+                            bucket: params.bucket,
+                            key: params.key,
+                            size: object_info_hub.size,
+                            content_type: params.content_type,
+                            xattr: object_info_hub.xattr,
+                            last_modified_time: (new Date(object_info_hub.create_time)).getTime(),
+                        };
+                        this.namespace_nb.upload_object(upload_params, object_sdk);
```

Comment:
should we return this promise back from the submit handler?

Reply:
Yes

Reply:
@guymguym Resolved and rebased.